### ISSUE TITLE
Breaks `const unsafe` functions

### ISSUE THREAD

**Issue by phil-opp** [rust-lang-nursery/rustfmt#852]
Rustfmt (version 0.3.0) changes the order of `const` and `unsafe` from the valid

```rust
pub const unsafe fn new() -> SomeType {..}
```

to the illegal

```rust
pub unsafe const fn new() -> SomeType {..}
```

**Comment by DarkDrek** [comment]
Seems to be the same as #850 and should be fixed after PR #851 is merged

**Comment by phil-opp** [comment]
Thanks! Closing as duplicate.
